sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/ManagementChannel.java

```diff
@@ -566,4 +640,38 @@ private Message createManagementMessage(String operation, String associatedLinkN
     private AmqpErrorContext getErrorContext() {
         return new SessionErrorContext(fullyQualifiedNamespace, entityPath);
     }
+
+    /**
+     * Get {@link RuleProperties} from message body.
+     *
+     * @param messageBody A message body which is {@link AmqpValue} type.
+     * @return A collection of {@link RuleProperties}.
+     *
+     * @throws AmqpException Get @{@link RuleProperties} from message body failed.
+     */
+    private List<RuleProperties> getRuleProperties(AmqpValue messageBody) {
+        try {
+            if (messageBody == null) {
+                return Collections.emptyList();
+            }
+
+            @SuppressWarnings("unchecked") List<Map<String, DescribedType>> rules = ((Map<String, List<Map<String, DescribedType>>>) messageBody.getValue())
+                .get(ManagementConstants.RULES);
+            if (rules == null) {
+                return Collections.emptyList();
+            }
+
+            List<RuleProperties> ruleProperties = new ArrayList<>();
+            for (Map<String, DescribedType> rule : rules) {
+                DescribedType ruleDescription = rule.get(ManagementConstants.RULE_DESCRIPTION);
+                ruleProperties.add(MessageUtils.decodeRuleDescribedType(ruleDescription));
+            }
+
+            return ruleProperties;
+        } catch (RuntimeException ex) {
+            throw logger.logExceptionAsError(Exceptions.propagate(new AmqpException(true,
+                String.format("Get rules failed. %s", ex.getMessage()),
+                getErrorContext())));
```

Comment:
I wonder if we really have to wrap this terminal exception as a retriable AmqpException?, or just throwing as it is is good enough \\cc @conniey 

(The reason for the mapping exception in ReactorDispatcher to AmqpException was the library to retry internally)

Reply:
I don't think we have to wrap this exception.